<?xml version="1.0" encoding="UTF-8" ?>

<launch>

    <!-- joy node -->
    <node respawn="false" pkg="joy" type="joy_node" name="joy" >
        <param name="dev" type="string" value="/dev/input/js0" />
        <!-- <param name="deadzone" value="0.12" /> -->
        <param name="autorepeat_rate" value="10.0" />
    </node>

    <!-- Axes -->

    <!--<node pkg="vrep_ros_teleop" type="teleop.py" name="teleop">-->
        <!--<param name="axis_linear" value="1" type="int"/>-->
        <!--<param name="axis_angular" value="0" type="int"/>-->
        <!--<param name="scale_linear" value="1.0" type="double"/>-->
        <!--<param name="scale_angular" value="-1.0" type="double"/>-->
        <!--<remap from="/teleop/joy" to="/joy"/>-->
    <!--</node>-->

    <node pkg="topic_tools" type="mux" name="cmd_mux"
          args="/rover/twistCommand /teleop/twistCommand /mux/autoCommand">
    </node>

    <node pkg="vrep_ros_teleop" type="teleop_mux.py" name="teleop_mux">
        <param name="joystick_button" value="0" type="int"/>
        <param name="joystick_topic" value="/teleop/twistCommand"/>
        <param name="auto_button" value="1" type="int"/>
        <param name="auto_topic" value="/mux/autoCommand"/>
        <remap from="/teleop_mux/joy" to="/joy"/>
    </node>

    <node pkg="rover_driver" type="rover_teleop.py" name="teleop">
        <param name="axis_linear_x" value="1" type="int"/>
        <param name="axis_linear_y" value="0" type="int"/>
        <param name="axis_angular" value="3" type="int"/>
        <param name="scale_linear" value="1.0" type="double"/>
        <param name="scale_angular" value="0.7" type="double"/>
        <remap from="/teleop/joy" to="/joy"/>
        <!--<remap from="/teleop/twistCommand" to="/rover/twistCommand"/>-->
    </node>

    <node pkg="rover_driver" type="rover_command.py" name="rover" output="screen">
        <param name="rover_name" value="rover"/>
        <param name="skidsteer" value="false"/>
    </node>

    <node name="task_server" pkg="floor_nav" type="task_server" output="screen" >
        <!-- After install -->
        <!-- <param name="lib_path" value="$(find floor_nav)/tasks"/> -->
        <!-- While developing -->
        <param name="lib_path" value="$(find floor_nav)/../../devel/share/floor_nav/tasks"/>
        <param name="base_frame" value="Hokuyo"/>
        <param name="reference_frame" value="rover/body"/>
        <!-- <param name="reference_frame" value="map"/>-->
    </node>

</launch>

